client/app/components/TimeAgo.jsx

```diff
@@ -0,0 +1,100 @@
+import moment from 'moment';
+import { isNil } from 'lodash';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { Moment } from '@/components/proptypes';
+import { clientConfig } from '@/services/auth';
+
+const interactiveComponents = new Set();
+
+function updateInteractiveComponents() {
+  interactiveComponents.forEach(component => component.forceUpdate());
+  setTimeout(updateInteractiveComponents, 30 * 1000);
+}
+updateInteractiveComponents();
+
+function toMoment(value) {
+  if (isNil(value)) {
+    return null;
+  }
+  value = moment(value);
+  return value.isValid() ? value : null;
+}
+
+export class TimeAgo extends React.Component {
+  static propTypes = {
+    date: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.instanceOf(Date),
+      Moment,
+    ]),
+    interactive: PropTypes.bool,
+    placeholder: PropTypes.string,
+  };
+
+  static defaultProps = {
+    date: null,
+    interactive: true,
+    placeholder: '',
+  };
+
+  constructor(props) {
+    super(props);
+    this._value = toMoment(this.props.date);
+  }
+
+  componentDidMount() {
+    if (this.props.interactive) {
+      interactiveComponents.add(this);
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    this._value = toMoment(nextProps.date);
+    if (nextProps.interactive) {
+      interactiveComponents.add(this);
+    } else {
+      interactiveComponents.delete(this);
+    }
+    return true; // default behaviour
+  }
```

Comment:
Although this works well, I believe `shouldComponentUpdate`'s purpose is to allow to block rendering depending on a condition. Perhaps `componentDidUpdate` would be more fitting (using `this.props` instead of `nextProps`).

Reply:
AFAIK `componentDidUpdate` is called after component receive new props and re-render. Therefore I'd need to call `forceUpdate` from it (to re-render it again with new `this._value`), which will lead to infinite loop because I have no a good condition when to stop calling it.
`componentWillUpdate` is deprecated, but (if I understand it correctly) `componentWillUpdate` was called when `shouldComponentUpdate` returns `true`, so technically it's a proper place for this.